avogadro/quantumio/gaussiancube.cpp

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2010 Geoffrey R. Hutchison
+  Copyright 2013 Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+
+#include "gaussiancube.h"
+
+#include <avogadro/core/molecule.h>
+#include <avogadro/core/utilities.h>
+#include <avogadro/core/cube.h>
+
+#include <iostream>
+
+namespace Avogadro {
+namespace QuantumIO {
+
+GaussianCube::GaussianCube()
+{
+}
+
+GaussianCube::~GaussianCube()
+{
+}
+
+std::vector<std::string> GaussianCube::fileExtensions() const
+{
+  std::vector<std::string> extensions;
+  extensions.push_back("cube");
+  return extensions;
+}
+
+std::vector<std::string> GaussianCube::mimeTypes() const
+{
+  return std::vector<std::string>();
+}
+
+bool GaussianCube::read(std::istream &in, Core::Molecule &molecule)
+{
+  // Variables we will need
+  std::string line;
+  std::vector<std::string> list;
+
+  int nAtoms;
+  Avogadro::Vector3 min;
+  Avogadro::Vector3 spacing;
+  Avogadro::Vector3i dim;
+
+  // Gaussian Cube format is very specific
+
+  // Read and set name
+  getline(in, line);
+  std::string cubeName = line;
+
+  // Read and skip field title (we may be able to use this to setCubeType in the future)
+  getline(in, line);
+
+  // Next line contains nAtoms and m_min
+  in >> nAtoms;
+  for (auto i = 0; i < 3; ++i)
```

Comment:
We don't use auto, this is C++11 which is not a hard dependency yet. We should possibly rethink it, but not just for auto in loops.